dcap/types/src/certification_data.rs

```diff
@@ -16,14 +16,46 @@ type Result<T> = ::core::result::Result<T, Quote3Error>;
 /// Table 9 of
 /// <https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf>.
 #[derive(Debug, Clone, Hash, PartialEq, Eq)]
-pub struct CertificationData<'a> {
-    // The `Certification Data` field as described in the QuoteLibReference.
-    // The length of this *will* equal the `size` field as described in the
-    // QuoteLibReference
+pub enum CertificationData<'a> {
+    /// Contains the following data:
+    /// - Platform provisioning ID (PPID)
+    /// - CPU security version number (CPUSVN)
+    /// - Provisioning certification enclave security version number (PCESVN)
+    /// - Provisioning certification enclave ID (PCEID)
+    Ppid(&'a [u8]),
+
+    /// Contains the following data encrypted with RSA 2048:
+    /// - Platform provisioning ID (PPID)
+    /// - CPU security version number (CPUSVN)
+    /// - Provisioning certification enclave security version number (PCESVN)
+    /// - Provisioning certification enclave ID (PCEID)
+    PpidEncryptedRsa2048(&'a [u8]),
+
+    /// Contains the following data encrypted with RSA 3072:
+    /// - Platform provisioning ID (PPID)
+    /// - CPU security version number (CPUSVN)
+    /// - Provisioning certification enclave security version number (PCESVN)
+    /// - Provisioning certification enclave ID (PCEID)
+    PpidEncryptedRsa3072(&'a [u8]),
+
+    /// Contains the provisioning certification key (PCK) leaf certificate
+    Pck(&'a [u8]),
+
+    /// Contains the certificate chain for the provisioning certification key
+    /// (PCK).
+    PckCertificateChain(PckCertificateChain<'a>),
+
+    /// ECDSA signature auxiliary data of an Intel SGX quote
+    /// See `sgx_ql_cert_key_type_t::ECDSA_SIG_AUX_DATA`
+    EcdsaSignatureAuxData(&'a [u8]),
+
+    /// Platform manifest
+    PlatformManifest(&'a [u8]),
+}
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
```

Comment:
This should have docs (and the crate should have `warn(missing_docs)` if it doesn't already). This can happen in a follow-up PR.